cmd2/cmd2.py

```diff
@@ -424,8 +433,13 @@ def __init__(self, completekey: str = 'tab', stdin=None, stdout=None, persistent
         # Used load command to store the current script dir as a LIFO queue to support _relative_load command
         self._script_dir = []
 
-        # Used when piping command output to a shell command
-        self.pipe_proc = None
+        # A flag used to protect critical sections in the main thread from stopping due to a KeyboardInterrupt
+        self.sigint_protection = utils.ContextFlag()
+
+        # If the current command created a process to pipe to, then this will be a ProcReader object.
+        # Otherwise the value will be None. This member is used to know when a pipe process can be killed
```

Comment:
Recommend trying to shorten this comment up a little bit.  Maybe something like:
```sh
# If the current command created a process to pipe to, then this will be a ProcReader object.
# Otherwise it will be None. Its used to know when a pipe process can be killed and/or waited upon.
```